lib/r10k/git/stateful_repository.rb

```diff
@@ -62,4 +61,13 @@ def status
       :insync
     end
   end
+
+  private
+
+  def sync?
+    return true if !@cache.exist?
+    return true if !(sha = @cache.resolve(@ref))
```

Comment:
Attempt to resolve a reference; if the ref can't be resolved we return true, hoping that fetching the cache will make the ref resolvable. This is also doing an inline assignment so we can look up the type of ref in the next line. Is this too abstruse?